cas-server-core-services/src/test/java/org/apereo/cas/services/AbstractResourceBasedServiceRegistryDaoTests.java

```diff
@@ -0,0 +1,391 @@
+        final RegisteredService r3 = this.dao.findServiceById(r2.getId());
+
+        assertEquals(r, r2);
+        assertEquals(r2, r3);
+        assertNotNull(r3.getAttributeReleasePolicy());
+        assertEquals(r2.getAttributeReleasePolicy(), r3.getAttributeReleasePolicy());
+    }
+
+    @Test
+    public void verifySaveMethodWithExistingServiceNoAttribute() {
+        final RegexRegisteredService r = new RegexRegisteredService();
+        r.setName("testSaveMethodWithExistingServiceNoAttribute");
+        r.setServiceId("testId");
+        r.setTheme("theme");
+        r.setDescription("description");
+        this.dao.save(r);
+        r.setTheme("mytheme");
+
+        this.dao.save(r);
+
+        final RegisteredService r3 = this.dao.findServiceById(r.getId());
+        assertEquals(r, r3);
+    }
+
+    @Test
+    public void verifySaveAttributeReleasePolicyMappingRules() {
+        final RegexRegisteredService r = new RegexRegisteredService();
+        r.setName("testSaveAttributeReleasePolicyMappingRules");
+        r.setServiceId("testId");
+
+        final Map<String, String> map = new HashMap<>();
+        map.put("attr1", "newattr1");
+        map.put("attr2", "newattr2");
+        map.put("attr2", "newattr3");
+
+
+        final ReturnMappedAttributeReleasePolicy policy = new ReturnMappedAttributeReleasePolicy();
+        policy.setAllowedAttributes(map);
+        r.setAttributeReleasePolicy(policy);
+
+        final RegisteredService r2 = this.dao.save(r);
+        final RegisteredService r3 = this.dao.findServiceById(r2.getId());
+
+        assertEquals(r, r2);
+        assertEquals(r2, r3);
+        assertNotNull(r3.getAttributeReleasePolicy());
+        assertEquals(r2.getAttributeReleasePolicy(), r3.getAttributeReleasePolicy());
+    }
+
+    @Test
+    public void verifySaveAttributeReleasePolicyAllowedAttrRules() {
+        final RegexRegisteredService r = new RegexRegisteredService();
+        r.setName("testSaveAttributeReleasePolicyAllowedAttrRules");
+        r.setServiceId("testId");
+
+        final ReturnAllowedAttributeReleasePolicy policy = new ReturnAllowedAttributeReleasePolicy();
+        policy.setAllowedAttributes(Arrays.asList("1", "2", "3"));
+        r.setAttributeReleasePolicy(policy);
+
+        final RegisteredService r2 = this.dao.save(r);
+        final RegisteredService r3 = this.dao.findServiceById(r2.getId());
+
+        assertEquals(r, r2);
+        assertEquals(r2, r3);
+        assertNotNull(r3.getAttributeReleasePolicy());
+        assertEquals(r2.getAttributeReleasePolicy(), r3.getAttributeReleasePolicy());
+    }
+
+    @Test
+    public void verifySaveAttributeReleasePolicyAllowedAttrRulesAndFilter() {
+        final RegexRegisteredService r = new RegexRegisteredService();
+        r.setName("testSaveAttributeReleasePolicyAllowedAttrRulesAndFilter");
+        r.setServiceId("testId");
+        r.setTheme("testtheme");
+        r.setEvaluationOrder(1000);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['1000' is a magic number.](https://www.codacy.com/app/mmoayyed/cas/file/2643620500/issues/source?bid=3256115&fileBranchId=3322154#l214)